### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by Georjane**
Created the game logic;
Split the code in main, board, and game.
The main is just the initial setup and game main loop;
The Board controls all the board information
The Game checks all turns, control the movements and winning conditions.

### CODE REVIEW

**Review by adaorachi — CHANGES_REQUESTED** [review]
### **Hi @Georjane and @fdfragoso :wave:** 

Well done on the completion of the first draft of this milestone. The game behavior works fine, however the structure of how the project should be is not exactly followed.

---
### CHANGES REQUIRED :recycle:

---

- [ ] You should encapsulate the functions you have inside the main.rb into classes and put them in the lib directory. And remember that the project requirement states that all game-logic related resources (classes and/or modules) should be kept in the lib directory [screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png)

- [ ] Your project should use at least two classes, e.g - Game class, Board class. And also ensure the game logic are divided between/among these classes.

- [ ] Since you will be calling some logic functions from the classes in the lib folder inside the main.rb/other logic files, you are allowed to use the getter/setter methods - attr_readers/attr_writers, and attr_accessor (but should only be used if it really needed to avoid its value from been mutated).

- [ ] You should ensure you have `puts/print` and `get.chomp` **ONLY** in the `main.rb` file

- [x] You should also ensure you have linter enabled for this project. Currently, you do not have the required Github action folder/files in the project root directory.

---

### After implementing the requirements stated above, you can request for another code review by submitting a code review form.

---

**Should you have any question, please feel free to reach out to me on Slack with the username @adaorachi**

Happy coding :relaxed: 

**MaryAnn Chukwuka**

File: bin/main [hunk 1]
```diff
@@ -24,113 +19,24 @@ def display_board(board)
...
-
-# THE POSITION IS ALREADY TAKEN
-def position_taken?(board, index)
-  if ((board[index]).is_a? Integer) || board[index].nil?
-    false
-  else
-    true
-  end
-end
+# game_on = true
```
Comment (uryelah) [inline bin/main]: - [x] Please delete commented out code and add the .rb extension to the main file. 
Make sure your code is well organized before submiting for a review.

File: lib/game.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    # rubocop: disable Style/GuardClause
+    if obj_board.valid_move?(board, index)
+      obj_board.move(board, index, current_player(board))
+    else
+      return ['Not a valid move!!! Try again']
+    end
+    # rubocop: disable Style/RedundantReturn
+    return board
+    # rubocop: enable Style/RedundantReturn
+    # rubocop: enable Style/GuardClause
```
Comment (uryelah) [inline lib/game.rb:28]: - [x] Don't overuse disable rubocop comments.
It's fairly easy and good practice to follow the ruby language style, such as avoiding redundant returns and using guard clauses. Read up on the error you got before disabling them and refactor your code to follow the ruby style best practices.

**Review by uryelah — CHANGES_REQUESTED** [review]
Hi **Jane Witah & Felipe Dacal Fragoso**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job creating your classes and modularizing your code a bit more. There are some issues to fix still, but read the comments below with care, make sure the changes are there, and you'll be good to go! 💪 

# Required Changes 🏗️ 🐛 💡

- [x] Improve your PR message, capitalize it, check the grammar describe what you did in this milestone, etc.

- [x] Make sure to pay attention to your commits, capitalize them and use the imperative tense.

**Comment by uryelah** [comment]
-----

 **Kindly fix the pointed out issues before requesting the next code review**

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: 

Thank you for applying the suggested changes.
You can merge :+1: 

**PS**: Please, kindly make `bin/main.rb` executable.

Good continuation :handshake: 

Mael FOSSO

**Pull request merged**